src/game.cpp

```diff
@@ -4801,137 +4803,90 @@ faction *game::list_factions(std::string title)
-void game::list_missions()
+void init_missions_tab( tabbed_window &win, input_broadcaster &ib, std::string &&tab_name, std::vector<mission *> &&data, std::string &&if_empty, bool can_confirm )
 {
-    WINDOW *w_missions = newwin(FULL_SCREEN_HEIGHT, FULL_SCREEN_WIDTH,
-                                (TERMY > FULL_SCREEN_HEIGHT) ? (TERMY - FULL_SCREEN_HEIGHT) / 2 : 0,
-                                (TERMX > FULL_SCREEN_WIDTH) ? (TERMX - FULL_SCREEN_WIDTH) / 2 : 0);
-
-    int tab = 0;
-    size_t selection = 0;
-    input_context ctxt("MISSIONS");
-    ctxt.register_cardinal();
-    ctxt.register_action("CONFIRM");
-    ctxt.register_action("QUIT");
-    ctxt.register_action("HELP_KEYBINDINGS");
-    while (true) {
-        werase(w_missions);
-        std::vector<mission*> umissions;
-        switch (tab) {
-        case 0:
-            umissions = u.get_active_missions();
-            break;
-        case 1:
-            umissions = u.get_completed_missions();
-            break;
-        case 2:
-            umissions = u.get_failed_missions();
-            break;
-        }
+    auto tab = win.create_tab( tab_name );
 
-        for (int i = 1; i < FULL_SCREEN_WIDTH - 1; i++) {
-            mvwputch(w_missions, 2, i, BORDER_COLOR, LINE_OXOX);
-            mvwputch(w_missions, FULL_SCREEN_HEIGHT - 1, i, BORDER_COLOR, LINE_OXOX);
+    constexpr size_t list_width = 30;
+    const size_t list_height = tab->get_rect().size_y;
+    constexpr auto draw_lambda = []( nc_color color, mission *miss )
+        {
+            auto name = miss->name();
 
-            if (i > 2 && i < FULL_SCREEN_HEIGHT - 1) {
-                mvwputch(w_missions, i, 0, BORDER_COLOR, LINE_XOXO);
-                mvwputch(w_missions, i, 30, BORDER_COLOR, LINE_XOXO);
-                mvwputch(w_missions, i, FULL_SCREEN_WIDTH - 1, BORDER_COLOR, LINE_XOXO);
+            nc_color col = color;
+            if( g->u.get_active_mission() == miss ) {
+                col = c_ltgreen;
             }
-        }
 
-        draw_tab(w_missions, 7, _("ACTIVE MISSIONS"), (tab == 0) ? true : false);
-        draw_tab(w_missions, 30, _("COMPLETED MISSIONS"), (tab == 1) ? true : false);
-        draw_tab(w_missions, 56, _("FAILED MISSIONS"), (tab == 2) ? true : false);
+            return std::pair<std::string, nc_color>{name, col};
+        }; // get name from mission, highlight active mission.
 
-        mvwputch(w_missions, 2, 0, BORDER_COLOR, LINE_OXXO); // |^
-        mvwputch(w_missions, 2, FULL_SCREEN_WIDTH - 1, BORDER_COLOR, LINE_OOXX); // ^|
+    auto _list = tab->create_child<ui_vertical_list<mission *, decltype(draw_lambda), nullptr>>( list_width, list_height, 0, 0, ui_anchor::top_left, draw_lambda );
+    _list->set_items( data );
 
-        mvwputch(w_missions, FULL_SCREEN_HEIGHT - 1, 0, BORDER_COLOR, LINE_XXOO); // |
-        mvwputch(w_missions, FULL_SCREEN_HEIGHT - 1, FULL_SCREEN_WIDTH - 1, BORDER_COLOR, LINE_XOOX); // _|
+    ib.subscribe( _list->default_action_handler() );
```

Comment:
This adds a listener for each of the three vertical lists to the input context. Pressing the scroll key scrolls in all three lists at once. (Test this with a character that has several completed *and* several failed missions.)

There is a strange double vertical border on the left side: one is from the tabbed window (and has nice corners), the other is directly adjacent and has no corners, like the line dividing the list from the description text. Is that intended?